sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -32,13 +35,17 @@ class MapboxMap internal constructor(
   MapFeatureQueryDelegate,
   ObservableInterface,
   MapListenerDelegate,
+  MapPluginExtensionsDelegate,
   MapStyleStateDelegate {
 
   private val nativeMapWeakRef = WeakReference(nativeMap)
   internal lateinit var style: Style
 
   private var terrainEnabled = false
 
+  private var cameraAnimationsPlugin: WeakReference<CameraAnimationsPlugin>? = null
+  private var gesturesPlugin: WeakReference<GesturesPlugin>? = null
```

Comment:
Since we can get the CameraAnimationsPlugin and GesturesPlugin from the MapView, would it possible to get a reference of the MapView inside the MapboxMap? and then we don't need such complex setup, instead, we could use an extension like:

```
interface MapViewProviderDelegate {
  fun getMapView() : MapView
}
```

`MapboxMap : MapViewProviderDelegate {
  ...
}`

And in the CameraAnimationsPlugin, we could have:

```
fun MapViewProviderDelegate.flyTo(point: Point) {
  this.getMapView().getCameraAnimationPlugin().flyTo(point)
}
```

wdyt?

Reply:
The main problem at all is that technically `MapView` contains `MapboxMap` : `MapView -> MapController -> MapboxMap`. If we add direct reverse link we technically get clear circular dependency.

Current approach seems lesser evil to me.

Reply:
I follow @pengdev suggestion of adding another interface, another abstraction layer, to work around the current issue. MapView might not be the best object as it's part of the SDK module not base. Abstractions need to occur in SDK base and be implemented in the SDK module to have clear separations of concerns. 

> circular dependency

In JVM languages, circular dependencies aren't as a big issue as with others languages. Don't get me wrong, it's a not a great practice but afaik the downsides are that they make the code hard to understand and make it difficult for the garbage collector to do his work. However, there is always a way to untangle these constructs by introducing a third party inside that circular relation.